packages/worker/src/worker.ts

```diff
@@ -854,85 +856,79 @@ export class Worker {
+                        `Got start event for an already running activity: ${base64TaskToken}`
+                      );
+                    }
+                    const info = await extractActivityInfo(
+                      task,
+                      this.options.loadedDataConverter,
+                      this.options.namespace,
+                      this.options.taskQueue
+                    );
+
+                    const { activityType } = info;
+                    const fn = this.options.activities.get(activityType);
+                    if (typeof fn !== 'function') {
+                      throw ApplicationFailure.create({
+                        type: 'NotFoundError',
+                        message: `Activity function ${activityType} is not registered on this Worker, available activities: ${JSON.stringify(
+                          [...this.options.activities.keys()]
+                        )}`,
+                        nonRetryable: false,
+                      });
+                    }
+                    let args: unknown[];
+                    try {
+                      args = await decodeArrayFromPayloads(this.options.loadedDataConverter, task.start?.input);
+                    } catch (err) {
+                      throw ApplicationFailure.fromError(err, {
+                        message: `Failed to parse activity args for activity ${activityType}: ${errorMessage(err)}`,
+                        nonRetryable: false,
+                      });
+                    }
+                    const headers = task.start?.headerFields ?? {};
+                    const input = {
+                      args,
+                      headers,
                     };
+
+                    this.logger.trace('Starting activity', activityLogAttributes(info));
+
+                    activity = new Activity(
+                      info,
+                      fn,
+                      this.options.loadedDataConverter,
+                      (details) =>
+                        this.activityHeartbeatSubject.next({
+                          type: 'heartbeat',
+                          info,
+                          taskToken,
+                          base64TaskToken,
+                          details,
+                          onError() {
+                            activity?.cancel('HEARTBEAT_DETAILS_CONVERSION_FAILED'); // activity must be defined
+                          },
+                        }),
+                      this.logger,
+                      this.options.interceptors.activity
+                    );
+                    output = { type: 'run', activity, input };
                     break;
-                  }
-                  let args: unknown[];
-                  try {
-                    args = await decodeArrayFromPayloads(this.options.loadedDataConverter, task.start?.input);
-                  } catch (err) {
+                  } catch (e) {
```

Comment:
Can we just move the break after the catch, and just have one `break`

Reply:
That's arguable, but intentional from my part. I personally feel that duplicating the break statement here makes it more obvious that we reached the conclusion of these logic branches. The `output = { ... }` statements don't stand out enough by themselves.